_episodes/23-continuous-integration-automated-testing.md

```diff
@@ -236,4 +236,43 @@ You should talk to your system administrator to setup dedicated runners for you.
 You can also use gitlab.com's runners for free if you provide your credit/debit card details,
 read more [here](https://about.gitlab.com/pricing/#why-do-i-need-to-enter-credit-debit-card-details-for-free-compute-minutes)
 
+> ## Exercise (Optional): Install self-managed GitLab runner.
+> Please refer to [this guide for various choices on how to install a local runner](https://docs.gitlab.com/runner/install/docker.html)
+>  
+> These are the steps that will work for most setups:
+> 
+> ### 0. Make sure you have docker installed on your machine.
+> [TNO Coding guild Docker guide](https://codingguild.tno.nl/coding-at-tno/programming-environments/?h=docker#docker (VPN needed))
+> 
+> ### 1. Install the GitLab runner through docker 
+> Optional: In MacOS systems, /srv does not exist by default. Create /private/srv, or another private directory, for setup.
```

Comment:
This might be confusing. If MacOS users work now work in `/private/srv`, they also need to adapt the code below. Also, `mkdir /private/srv` require `sudo` permissions. I would suggest to use `$HOME/srv` and add a note regarding the following commands:
```suggestion
> **Note**: In MacOS systems, /srv does not exist by default. Create `$HOME/srv` (with `mkdir $HOME/srv`), or another private directory, for setup.
> This requires that you also update the following command. Replace `/srv` with `$HOME/srv`.
```
Also, it is not really optional for MacOS users, so I would remove that.

Reply:
Yes, definitely agree and I really like your suggestion! In general, this 'exercise' will be a little adventure for everyone so requires our attention in guiding participants through it. For example, I had to use a docker volume because somehow could not figure out how to use the local folder in `$HOME/srv`.